pkg/bgpv1/logger/log.go

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package logger
+
+import (
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var (
+	// Log to be used in BGP package. All log messages emitted by this have k/v
+	// `subsys=bgp-control-plane`.
+	Log = logging.DefaultLogger.WithField(logfields.LogSubsys, "bgp-control-plane")
```

Comment:
You can DI this logger through Hive. You can inject the type `logrus.FieldLogger`. The logger already contains the field `subsys=<cell module name>`.

Reply:
Taking this approach - as all the reconcilers are cells -it's easy to inject logger this way. Will move log field KVs to types package.

Reply:
Yep, notice that the subsys is the name of the Module, so when we DI this to reconcilers, we still have `subsys=bgp-cp`. I would add a field `reconciler=<pod-cidr|service|pod-ippool|etc...>` for each reconciler to make filtering easier.

Reply:
Added new reconciler log field.